features/FEATURE_LWIP/lwip-interface/EthernetInterface.h

```diff
@@ -100,6 +100,18 @@ class EthernetInterface : public EthInterface
      */
     virtual const char *get_gateway();
 
+    /** Register callback for status reporting
+     *
+     *  @param status_cb The callback for status changes
+     */
+    virtual void register_status_callback(mbed::Callback<void(connection_status_t, int)> status_cb);
```

Comment:
Still not answered, is this regular way how to register a callback? many API in mbed has `attach`, shall this be also aligned?

Reply:
What about how to register callback? I would like to have an alignment if possible or an argument why this is register_status_callback_name